src/isar/state_machine/states/maintenance.py

```diff
@@ -18,7 +19,13 @@ def _release_from_maintenance_handler(event: Event[bool]):
                 events.api_requests.release_from_maintenance_mode.response.trigger_event(
                     True
                 )
-                return state_machine.release_from_maintenance  # type: ignore
+
+                robot_status = state_machine.shared_state.robot_status.check()
+                if robot_status == RobotStatus.Home:
+                    return state_machine.goto_home  # type: ignore
```

Comment:
Tbf it will anyways go to intervention needed if it goes to home without having status home, I think